handlers/svg-depth-map/svg-depth-map.py

```diff
@@ -0,0 +1,174 @@
+        definitions_schema = json.load(f)
+    with open("./schemas/request.schema.json") as f:
+        request_schema = json.load(f)
+    with open("./schemas/handler-response.schema.json") as f:
+        response_schema = json.load(f)
+    with open("./schemas/renderers/svglayers.schema.json") as f:
+        renderer_schema = json.load(f)
+    store = {
+        definitions_schema["$id"]: definitions_schema,
+        request_schema["$id"]: request_schema,
+        response_schema["$id"]: response_schema,
+        renderer_schema["$id"]: renderer_schema,
+    }
+    resolver = jsonschema.RefResolver.from_schema(
+        request_schema, store=store
+    )
+    # Get and validate request contents
+    contents = request.get_json()
+    try:
+        validator = jsonschema.Draft7Validator(
+            request_schema, resolver=resolver
+        )
+        validator.validate(contents)
+    except ValidationError as e:
+        logging.error(e)
+        return jsonify("Invalid request received!"), 400
+
+    # Check preprocessor data
+    preprocessors = contents['preprocessors']
+    if ("ca.mcgill.a11y.image.capability.DebugMode"
+        not in contents['capabilities']
+            or "ca.mcgill.a11y.image.renderer.SVGLayers"
+            not in contents["renderers"]):
+        logging.debug("Debug mode inactive")
+        print("debug inactive")
+        response = {
+            "request_uuid": contents["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(
+                response_schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+
+    # No Object Detector found
+    if "ca.mcgill.a11y.image.preprocessor.depth-map-gen"\
+            not in preprocessors:
+        logging.debug("No Depth Map found")
+        response = {
+            "request_uuid": contents["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(
+                response_schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+    if "dimensions" in contents:
+        # If an existing graphic exists, often it is
+        # best to use that for convenience.
+        # see the following for SVG coordinate info:
+        # developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Positions
+        dimensions = contents["dimensions"]
+    else:
```

Comment:
It's unclear to me why the absence of dimensions here would be blocking. The debug data that you're working with is a raster image whose dimensions are (should be?) the same as the request. Anyway, no need to change since this field should be present, just a bit odd.